src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Adapter\ExtLdap;
+
+use Doctrine\DBAL\Exception\DriverException;
+use Symfony\Component\Ldap\Adapter\AbstractQuery;
+use Symfony\Component\Ldap\Exception\LdapException;
+
+/**
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class Query extends AbstractQuery
+{
+    /** @var  Connection */
+    protected $connection;
+
+    /** @var resource */
+    private $search;
+
+    /**
+     * Constructor.
```

Comment:
The problem is that PHPStorm actually does not infer correctly the type of the object, because of the Typehint in the parent constructor.